### PR TITLE
(feat) NVIDIA connectors update

### PR DESCRIPTION
**Pull request by raspawar**
# Description

This PR introduces significant improvements to the NVIDIA connectors, focusing on enhancing reliability, maintainability, and test coverage. The changes include structural improvements to the base URL validation logic, model client validation, and a new Model class implementation.

### Base URL and Model Client Validation
- Enhanced base URL validation to ensure proper formatting and error handling
- Improved model client validation with better error messages
- Added type hints and docstrings for better code clarity
- Implemented robust error handling for invalid configurations

### Code Structure Improvements
- Created a dedicated Model class to handle model-specific logic
- Refactored embedding implementation for better maintainability
- Updated LLM connector with improved type safety
- Enhanced reranker implementation with better validation

### Testing Enhancements
- Added comprehensive test suite for URL validation scenarios
- Implemented new test cases for embedding functionality
- Added tests for LLM connector edge cases
- Included reranker test coverage
- Updated existing tests to match new implementations

## Type of Change
- [x] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Testing
All new and existing tests pass locally. Added new test cases to cover:
- URL validation edge cases
- Model client initialization scenarios
- Embedding functionality with different inputs
- LLM connector with various configurations
- Reranker functionality with different parameters

## Additional Notes
- This update improves the overall reliability of NVIDIA connectors
- Enhanced error messages will help users debug issues more effectively
- Better type hints and documentation make the code more maintainable

## Related Issues
- Fixes URL validation issues
- Improves connector reliability
- Enhances error handling

cc: @mattf @nreamaroon @dglogo @sumitb

### CODE REVIEW

File: llama-index-integrations/llms/llama-index-llms-nvidia/tests/test_api_key.py [hunk 1]
```diff
@@ -35,8 +35,7 @@ def get_api_key(instance: Any) -> str:
 
 
 def test_create_default_url_without_api_key(masked_env_var: str) -> None:
```
Comment (mattf) [inline llama-index-integrations/llms/llama-index-llms-nvidia/tests/test_api_key.py:37]: this is a test for behavior when api_key is not passed. the body is changed to something else.

File: llama-index-integrations/embeddings/llama-index-embeddings-nvidia/tests/test_embeddings_nvidia.py [hunk 2]
```diff
@@ -110,8 +110,7 @@ def test_model_incompatible_client_model() -> None:
 
 def test_model_incompatible_client_known_model() -> None:
     model_name = "google/deplot"
-    warn_msg = f"Unable to determine validity"
-    with pytest.warns(UserWarning) as msg:
```
Comment (mattf) [inline llama-index-integrations/embeddings/llama-index-embeddings-nvidia/tests/test_embeddings_nvidia.py:114]: this is a type change that may break user code

File: llama-index-integrations/llms/llama-index-llms-nvidia/tests/test_base_url.py [hunk 3]
```diff
@@ -32,27 +30,20 @@ def mock_local_models(httpx_mock: HTTPXMock, base_url: str):
...
-        "https://test_url/.../v1",
+        "http://0.0.0.0:8888/chat/completion",
+        "https://0.0.0.0:8888/ranking",
     ],
 )
-def test_base_url_invalid_not_hosted(base_url: str) -> None:
-    with pytest.raises(ValueError) as msg:
-        Interface(base_url=base_url)
-    assert "Invalid base_url" in str(msg.value)
+def test_base_url_invalid_not_hosted(base_url: str, mock_local_models: None) -> None:
```
Comment (mattf) [inline llama-index-integrations/llms/llama-index-llms-nvidia/tests/test_base_url.py:37]: this changes semantics where being invalid should raise an error

File: llama-index-integrations/llms/llama-index-llms-nvidia/tests/test_base_url.py [hunk 4]
```diff
@@ -32,27 +30,20 @@ def mock_local_models(httpx_mock: HTTPXMock, base_url: str):
...
+@pytest.mark.parametrize("base_url", ["http://localhost:8080/v1"])
 def test_base_url_valid_not_hosted(base_url: str, mock_local_models: None) -> None:
     with pytest.warns(UserWarning):
         Interface(base_url=base_url)
 
 
 @pytest.mark.parametrize("base_url", ["https://integrate.api.nvidia.com/v1/"])
 def test_base_url_valid_hosted(base_url: str) -> None:
-    Interface(base_url=base_url)
+    Interface(base_url=base_url, api_key="BOGUS")
```
Comment (mattf) [inline llama-index-integrations/llms/llama-index-llms-nvidia/tests/test_base_url.py:49]: check on the change that requires specifying the api_key

File: llama-index-integrations/postprocessor/llama-index-postprocessor-nvidia-rerank/tests/test_postprocessor_nvidia_rerank.py [hunk 5]
```diff
@@ -226,8 +226,6 @@ def test_model_incompatible_client() -> None:
 
 def test_model_incompatible_client_known_model() -> None:
     model_name = "google/deplot"
-    warn_msg = f"Unable to determine validity"
-    with pytest.warns(UserWarning) as msg:
+    with pytest.raises(ValueError) as msg:
```
Comment (mattf) [inline llama-index-integrations/postprocessor/llama-index-postprocessor-nvidia-rerank/tests/test_postprocessor_nvidia_rerank.py:229]: this indicates a type change that could break user code

**Review by mattf — COMMENTED** [review]
please check on the api changes that may break user code

File: llama-index-integrations/postprocessor/llama-index-postprocessor-nvidia-rerank/tests/test_postprocessor_nvidia_rerank.py (hunk 5 above)
Comment (raspawar) [reply llama-index-integrations/postprocessor/llama-index-postprocessor-nvidia-rerank/tests/test_postprocessor_nvidia_rerank.py:229]: yes I will revert that asup

File: llama-index-integrations/embeddings/llama-index-embeddings-nvidia/tests/test_base_url.py [hunk 6]
```diff
@@ -25,19 +29,69 @@ def mock_local_models(httpx_mock: HTTPXMock, base_url: str):
     )
 
 
-# test case for base_url warning
+def test_create_without_base_url(public_class: type, monkeypatch) -> None:
+    monkeypatch.setenv("NVIDIA_API_KEY", "valid_api_key")
+    with no_env_var("NVIDIA_BASE_URL"):
```
Comment (mattf) [inline llama-index-integrations/embeddings/llama-index-embeddings-nvidia/tests/test_base_url.py:34]: try monkeypatch.delenv("NVIDIA_BASE_URL")

**Review by logan-markewich — APPROVED** [review]
